pages/science/reactors/extensiblereactor.rst

```diff
@@ -0,0 +1,160 @@
+the following state variables:
+
+- :math:`m`, the mass of the reactor's contents (in kg)
+
+- :math:`T`, the temperature (in K)
+
+- :math:`Y_k`, the mass fractions for each species (dimensionless)
+
+As shown in the derivations of the governing equations for an Ideal Gas 
+Constant Pressure Reactor, the user may modify the 3 equations below:
+
+.. math::
+
+   \frac{dm}{dt} = \sum_{in} \dot{m}_{in} - \sum_{out} \dot{m}_{out} +
+                    \dot{m}_{wall}
+                    \tag{1}
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+.. math::
+
+   m \frac{dY_k}{dt} = \sum_{in} \dot{m}_{in} (Y_{k,in} - Y_k)+
+                      \dot{m}_{k,gen} - Y_k \dot{m}_{wall}
+                      \tag{3}
+
+There are two "sides" to each of these equations: the terms left of the equals
+sign and the terms to the right of the equals sign. This is the format
+in which the user will be editing the governing equations. For example,
+if the user wishes to add a term for a large mass (say a rock) inside
+the reactor to see the effects on reaction temperature:
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+Will change to:
+
+.. math::
+
+   m c_p \frac{dT}{dt} + m_{rock} c_{p,rock} \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+
+A simple example is shown below to illustrate the process for implementing
+changes in Cantera's existing governing equations.
+We will be replacing the right-hand side (RHS) and left-hand side (LHS) of 
+the temperature governing equation for an Ideal Gas Constant Pressure Reactor.
+All other governing equations defining an Ideal Gas Constant Pressure Reactor
+will remain as the default.
+
+In this example
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+Will change to:
+
+.. math::
+
+   m_{rock} c_{p,rock}\frac{dT}{dt} + m_{gas}\frac{dT}{dt} = - \dot{Q}
+
+1. Define objects, properties, and initial conditions.
+
+.. code-block:: python
+
+   #create gas object
+   gas = ct.Solution('h2o2.yaml')
+   gas.TPX = 500, ct.one_atm, 'H2:2,O2:1,N2:4'
+   gas_initial_enthalpy = gas.enthalpy_mass
+
+   #define properties of gas and solid
+   mass_gas = 20 #[kg]
+   Q = 100 #[J/s]
+   mass_lump = 10 #[kg]
+   cp_lump = 1.0 #[J/kgK]
+
+   #initialize time at zero
+   time = 0 #[s]
+   n_steps = 300
+
+2. Define a new custom Reactor class. Here we named it "DummyReactor" and 
+chose the Ideal Gas Constant Pressure Reactor as the base class to inheret
+governing equations from. An Extensible Reactor is also known as a Delegated 
+Reactor.
```

Comment:
This last bit about extensible vs. delegated needs to be highlighted in a more prominent way, IMO. I missed it on first reading this page.